src/test/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtilTest.java

```diff
@@ -379,6 +391,36 @@ public void testEscapeAllControlChars() {
                 JavadocUtil.escapeAllControlChars("1\\r2\\n3\\t"), "invalid result");
     }
 
+    @Test
+    public void testGetContentOfInlineCustomTag() throws CheckstyleException, IOException {
+        final File testFile = new File("src/test/resources/com/"
+                + "puppycrawl/tools/checkstyle/utils/javadocutil/"
+                + "InputJavadocUtilTest_ContentOfInlineCustomTag.java");
+        final DetailAST root = JavaParser.parseFile(testFile, JavaParser.Options.WITH_COMMENTS);
+        final DetailAST javadocCommentAST = root.getLastChild().getFirstChild().getFirstChild()
+                .getNextSibling();
```

Comment:
I do not understand what is going on here. We parse file and then adding something to AST? Why not put everything in file?

Reply:
>We parse file and then adding something to AST? Why not put everything in file?

We are not adding something to the ast, the util method takes DetailNode of type `JavadocTokenTypes.JAVADOC_INLINE_TAG` as input, so we parse the file, get the root node, then from the root node we traverse to the required node and perform checking.

>If we need only ast node, why do we need a file in this case?

We need the file otherwise I would have to set text and everything else explicitly which won't be good imo.